cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -61,6 +101,38 @@ def plot(self, ax: Optional[plt.Axes] = None,
             fig.show()
         return ax
 
+    def depolarizing_model(self) -> CrossEntropyDepolarizingModel:
+        """Fit a depolarizing error model for a cycle.
+
+        Fits an exponential model f = S * p^d, where d is the number of cycles
+        and f is the cross entropy fidelity for that number of cycles,
+        using nonlinear least squares.
+
+        Returns:
+            A DepolarizingModel object, which has attributes `coefficient`
+            representing the value S, `decay_constant` representing the value
+            p, and `covariance` representing the covariance in the estimation
+            of S and p in that order.
+        """
+        # Get initial guess by linear least squares with logarithm of model
+        x = [depth for depth, fidelity in self.data if fidelity > 0]
+        y = [np.log(fidelity) for _, fidelity in self.data if fidelity > 0]
+        fit = np.polynomial.polynomial.Polynomial.fit(x, y, 1).convert()
+        p0 = np.exp(fit.coef)
+
+        # Perform nonlinear least squares
+        x = [depth for depth, _ in self.data]
+        y = [fidelity for _, fidelity in self.data]
+
+        def f(d, S, p):
+            return S * p**d
+
+        params, covariance = scipy.optimize.curve_fit(f, x, y, p0=p0)
+
+        return CrossEntropyDepolarizingModel(spam_depolarization=params[0],
+                                             cycle_depolarization=params[1],
+                                             covariance=covariance)
```

Comment:
Out of curiosity: Do you see non-zero off-diagonal entries in the covariance matrix in practice?